hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/interceptor/AbstractInterceptor.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.bytecode.enhance.spi.interceptor;
+
+import org.hibernate.engine.spi.PersistentAttributeInterceptor;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+
+/**
+ * @author Steve Ebersole
+ */
+public abstract class AbstractInterceptor implements PersistentAttributeInterceptor, SessionAssociableInterceptor {
+	private final String entityName;
+
+	private transient SharedSessionContractImplementor session;
+	private boolean allowLoadOutsideTransaction;
+	private String sessionFactoryUuid;
+
+	@SuppressWarnings("WeakerAccess")
+	public AbstractInterceptor(String entityName) {
+		this.entityName = entityName;
+	}
+
+	public String getEntityName() {
+		return entityName;
+	}
+
+	@Override
+	public SharedSessionContractImplementor getLinkedSession() {
+		return session;
+	}
+
+	@Override
+	public void setSession(SharedSessionContractImplementor session) {
+		this.session = session;
```

Comment:
Should further checks be done when  ```this.session != null && session != null```, similar to what is done in [```AbstractPersistentCollection#setCurrentSession```](https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/collection/internal/AbstractPersistentCollection.java#L678-L698)?

Reply:
It would not hurt I guess.  But again this was taken directly from the existing interceptor code.  Have you seen trouble with that?  I think this is another one of "not specific to this work"